calico_containers/pycalico/ipam.py

```diff
@@ -444,21 +450,49 @@ def _auto_assign(self, ip_version, num, handle_id,
         block_list = self._get_affine_blocks(hostname,
                                              ip_version,
                                              pool)
-        block_ids = iter(block_list)
+        block_ids = list(block_list)
+        block_ids.reverse()
+        key_errors = 0
         allocated_ips = []
 
         num_remaining = num
         while num_remaining > 0:
             try:
-                block_id = block_ids.next()
-            except StopIteration:
+                block_id = block_ids.pop()
```

Comment:
pop(0) could have been used instead of needing to reverse the list - but not fussed either way